docs/bundles.md

```diff
@@ -0,0 +1,151 @@
+# Bundles in The Charmstore
+
+The charmstore allows two versions of bundle specifications, as described by
+github.com/juju/charm.  The versions are numbered 3 and 4, relating to the API
+version under which they can be hosted: charmworld (API v3) supports only
+version 3 bundles, charmstore (API v4) supports version 3 and version 4.
+
+## Version 3 bundles
+
+Version 3 bundles are currently existing bundles that specify a deployment as a
+list of services and, optionally, relations.  The charmstore will not support
+the idea of a "basket" or multiple bundles within one file.  However, existing
+baskets will still be imported, and split up into their component bundles.
+
+## Version 4 bundles
+
+Version 4 bundles are identical to version 3 bundles except for a few key
+differences: the `branch` attribute of the service spec is no longer supported,
+they must contain a machine specification, and their deployment directives are
+different from version 3 bundles.  A bundle without a machine specification is
```

Comment:
Good point; I've moved this to a separate paragraph about deploying the bundles with the deployer and GUI.